internal/app/synchronizer/synchronizer_service.go

```diff
@@ -296,6 +291,11 @@ Registration:
 	})
 
 	<-closedOnCycleEnd
+	err := s.store.CleanupBackfills(ctx)
+	if err != nil {
+		logger.Errorf("Failed to clean up backfills, %s", err.Error())
+	}
+
 	stats.Record(ctx, iterationLatency.M(float64(time.Since(cst)/time.Millisecond)))
 
 	// Clean up in case it was never needed.
```

Comment:
What will happen if cancelProposalCollection fires before cleanup finishes its work?

Reply:
moved CleanupBackfills to the very end of the function